adapters/invibes/invibes.go

```diff
@@ -0,0 +1,327 @@
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/pbsmetrics"
+)
+
+const adapterVersion = "prebid_1.0.0"
+
+type InvibesAdRequest struct {
+	BidParamsJson string
+	Location      string
+	Lid           string
+	IsTestBid     bool
+	Kw            string
+	IsAmp         bool
+	Width         string
+	Height        string
+	GdprConsent   string
+	Gdpr          bool
+	Bvid          string
+	InvibBVLog    bool
+	VideoAdDebug  bool
+}
+type InvibesBidParams struct {
+	PlacementIds []string
+	BidVersion   string
+	Properties   map[string]InvibesPlacementProperty
+}
+type InvibesPlacementProperty struct {
+	Formats []openrtb.Format
+	ImpId   string
+}
+type InvibesInternalParams struct {
+	BidParams   InvibesBidParams
+	Host        string
+	IsAmp       bool
+	Gdpr        string
+	GdprConsent string
+
+	TestIp   string
+	TestBvid string
+	TestLog  bool
+}
+type BidServerBidderResponse struct {
+	Currency  string              `json:"currency"`
+	TypedBids []BidServerTypedBid `json:"typedBids"`
+	Error     string              `json:"error"`
+}
+type BidServerTypedBid struct {
+	Bid          openrtb.Bid `json:"bid"`
+	DealPriority int         `json:"dealPriority"`
+}
+
+func (a *InvibesInternalParams) IsTestRequest() bool {
+	return a.TestIp != "" || a.TestBvid != ""
+}
+
+type InvibesAdapter struct {
+	EndpointTemplate template.Template
+}
+
+func NewInvibesBidder(endpointTemplate string) *InvibesAdapter {
+	urlTemplate, err := template.New("endpointTemplate").Parse(endpointTemplate)
+	if err != nil {
+		glog.Fatal("Unable to parse endpoint url template")
+		return nil
+	}
+	return &InvibesAdapter{EndpointTemplate: *urlTemplate}
+}
+
+func (a *InvibesAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No imps in the bid request",
+		}}
+	}
+
+	consentString := ""
+	if request.User != nil {
+		var extUser openrtb_ext.ExtUser
+		if err := json.Unmarshal(request.User.Ext, &extUser); err == nil {
+			consentString = extUser.Consent
+		}
+	}
+	gdprApplies := "1"
+	var extRegs openrtb_ext.ExtRegs
+	if request.Regs != nil {
+		if err := json.Unmarshal(request.Regs.Ext, &extRegs); err == nil {
+			if extRegs.GDPR != nil && (*extRegs.GDPR == 0 || *extRegs.GDPR == 1) {
+				gdprApplies = strconv.Itoa(int(*extRegs.GDPR))
+			}
+		}
+	}
+
+	var httpRequests []*adapters.RequestData
+	var errors []error
+
+	var invibesInternalParams InvibesInternalParams = InvibesInternalParams{
+		BidParams: InvibesBidParams{
+			Properties: make(map[string]InvibesPlacementProperty),
+			BidVersion: "1",
+		},
+	}
+
+	for _, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: "Error parsing bidderExt object",
+			})
+		}
+		if len(errors) == 0 {
```

Comment:
I'm not sure this is doing what you intend. If there is an error with imp-1, then this statement would be false for every imp. I suspect you want to only continue parsing, if so consider adding a `continue` in the Unmarshal error handling branch above.